### PR TITLE
WIP - 457 fix studies for variants

### PR DESCRIPTION
**Pull request by lakinwecker**

### CODE REVIEW

File: modules/round/src/main/JsonView.scala [hunk 1]
```diff
@@ -38,9 +38,10 @@ final class JsonView(
   private def kingMoves(game: Game, playerIndex: PlayerIndex) =
     (game.variant.frisianVariant) option game.history.kingMoves(playerIndex)
 
-  private def onlyDropsVariantForCurrentAction(pov: Pov): Boolean = {
-    pov.game.variant.onlyDropsVariant || (pov.game.variant.key == "amazons" && pov.game.situation.destinations.size == 0)
-  }
+  private def onlyDropsVariantForCurrentAction(pov: Pov): Boolean = pov.game.variant.onlyDropsVariant
+  // TODO: in analysis mode, this will be evaluated against the last move, but we don't want to set onlyDropsVariant
+  // in this case.
+  // pov.game.variant.onlyDropsVariant || (pov.game.variant.key == "amazons" && pov.game.situation.destinations.size == 0)
```
Comment (lakinwecker) [inline modules/round/src/main/JsonView.scala:44]: I don't remember what this was for, I think I'll reevaluate it when I'm finalizing the amazons integration after the multimove stuff.

File: modules/study/src/main/StudySocket.scala [hunk 2]
```diff
@@ -119,8 +119,13 @@ final private class StudySocket(
           }
         case "shapes" =>
           reading[AtPosition](o) { position =>
-            (o \ "d" \ "shapes").asOpt[List[Shape]] foreach { shapes =>
-              who foreach api.setShapes(studyId, position.ref, Shapes(shapes take 32))
+            {
+              // TODO: this will only parse up to a 10x10 board
```
Comment (lakinwecker) [inline modules/study/src/main/StudySocket.scala:123]: This should be updated for go19x19 once it's merged.

File: modules/study/src/main/TreeBuilder.scala [hunk 3]
```diff
@@ -18,14 +18,19 @@ object TreeBuilder {
         val sit = Game(variant.gameLogic, variant.some, root.fen.some).situation
         sit.playable(false) ?? sit.destinations
       }
-    makeRoot(root, variant).copy(dests = dests.some)
+    val dropsByRole = {
+      val sit = Game(variant.gameLogic, variant.some, root.fen.some).situation
+      sit.playable(false) ?? sit.dropsByRole
+    }
```
Comment (lakinwecker) [inline modules/study/src/main/TreeBuilder.scala:24]: So this code only adds the drops if the situation is playable. I wonder if all of the places that we use the drops by role should do the same thing? Probably.

Comment (msomatt) [reply modules/study/src/main/TreeBuilder.scala:24]: Maybe in the context of study. I expect that there are other checks made in the case of round

File: ui/analyse/src/socket.ts [hunk 4]
```diff
@@ -127,6 +127,7 @@ export function make(send: AnalyseSocketSend, ctrl: AnalyseCtrl): Socket {
       // no strict equality here!
       if (data.ch == currentChapterId()) ctrl.addNode(data.node, data.path);
       else console.log('socket handler node got wrong chapter id', data);
+      console.log(`data.node.fen: ${data.node.fen}`);
```
Comment (msomatt) [inline ui/analyse/src/socket.ts:130]: probably want to remove this debugging?

File: modules/relay/src/main/RelayFormat.scala [hunk 5]
```diff
@@ -82,10 +84,13 @@ final private class RelayFormatApi(ws: StandaloneWSClient, cacheApi: CacheApi)(i
         case _                        => none
       }
 
-  private def looksLikePgn(body: String): Boolean =
+  private def looksLikePgn(body: String): Boolean = {
+    // TODO: Only support chess PGN for now.
```
[the remark this answers is not in the document]
Comment (msomatt) [reply modules/relay/src/main/RelayFormat.scala:88]: Broadcasts (see lichess.org/broadcast)

**Comment by JamesHeppell** [comment]
Released within another PR (fix studies?)

**Pull request closed without merging**